lint-spec.ts

```diff
@@ -173,14 +103,163 @@ function getReportedInputLocation(directory: SpecDirectory): string {
   }
 
   if (directory.hasFile('input.sass')) {
-    return getRelativePath(path.resolve(directory.path, 'input.sass'));
+    return getRelativePath(p.resolve(directory.path, 'input.sass'));
   }
 
-  return getRelativePath(path.resolve(directory.path, 'input.scss'));
+  return getRelativePath(p.resolve(directory.path, 'input.scss'));
 }
 
 function getRelativePath(filepath: string): string {
-  return path.relative(process.cwd(), filepath);
+  return p.relative(process.cwd(), filepath);
 }
 
-lintAllTests();
+/** The number of lines allowed in HRX files. */
+const LINE_THRESHOLD = 500;
+
+/**
+ * Verifies all HRX files in the {@link rootPath} directory are under the
+ * {@link LINE_THRESHOLD}.
+ *
+ * HRX files are unwrapped and transformed into real files when {@link fix} is
+ * `true`.
+ */
+async function lintHrxSize(
+  rootPath: string,
+  reporter: LintReporter,
+  {fix = false}
+) {
+  const rootDir = await fromPath(rootPath);
+  const tooBig = await bigHrxInTree(rootDir, reporter);
```

Comment:
Does it still make sense for this function to do its own recursive traversal of the test tree separate from `lintDirectory()`?

Reply:
The current implementation of `lintDirectory()` + `lintNestedTestDirectories()` makes it slightly awkward to merge `lintHrxSize()` into the `lintDirectory()` because it stops traversing when the directory `isTestDir` but the HRX size linter wants to keep traversing down

IMO this needs to either be refactored to separate traversal and linting or keep current implementation of each "sub-linter" running their own traversal

I think it makes sense to leave it as-is for now, and think about refactoring when another "linter" comes by